powerstore/datasource_volume_group.go

```diff
@@ -258,7 +266,19 @@ func (d *volumeGroupDataSource) Read(ctx context.Context, req datasource.ReadReq
 		volumeGroup, err = d.client.PStoreClient.GetVolumeGroup(context.Background(), state.ID.ValueString())
 		volumeGroups = append(volumeGroups, volumeGroup)
 	} else {
-		volumeGroups, err = d.client.PStoreClient.GetVolumeGroups(context.Background())
+		err = validateVolumeFilter(state.Filters.ValueString())
+		if err != nil {
+			resp.Diagnostics.AddError(
+				"Invalid filter expression",
+				err.Error(),
+			)
```

Comment:
Please add attribute errors here, so that users get better error messages.
```suggestion
			resp.Diagnostics.AddAttributeError(
				path.Root("filter"),
				"Invalid filter expression",
				err.Error(),
			)
```
Applicable to similar scenarios.

Reply:
Addressed in recent commit